shm/src/Store.cpp

```diff
@@ -0,0 +1,248 @@
+                                                    amount,
+                                                    fruit->getBasePrice()));
+    } else if (typeid(cargo) == typeid(Item)) {
+        Item* item = static_cast<Item*>(cargo);
+        cargo_.push_back(std::make_unique<Item>(item->getName(),
+                                                amount,
+                                                item->getBasePrice(),
+                                                item->getRarity()));
+    }
+}
+
+void Store::nextDay() {
+    for (const auto& cargo : cargo_) {
+        cargo->setAmount(randomGenerate(MIN_CARGO_IN_STORE, MAX_CARGO_IN_STORE));
+    }
+}
+
+size_t Store::randomGenerate(size_t min, size_t max) {
+    std::mt19937 gen(std::random_device{}()); 
+    std::uniform_int_distribution<size_t> distribution(min, max);
+    return distribution(gen);
+}
+
+Item::Rarity Store::rarityConversion(size_t numberForConversion) {
+    size_t multiplyRarity = 4;
+    size_t addRarity = 1;
+    return static_cast<Item::Rarity>((numberForConversion * multiplyRarity) + addRarity);
+}
+
+void Store::generateAllCargo() {
+    std::string line;
+    std::fstream file;
+    file.open("settings/items.txt", std::ios::in);
+    if (file.fail()) {
+        std::cout << "File not exist store\n";
+    }
+    while (getline(file, line)) {
+        convertDataFromFile(line);
+    }
+    file.close();
+}
+
+void Store::convertDataFromFile(std::string lineFromFile) {
+    std::string singleWordFromFile{};
+    size_t wordNumber{};
+    for (size_t i = 0; i <= lineFromFile.size(); i++) {
+        if (lineFromFile[i] != ' ' && i < lineFromFile.size()) {
+            singleWordFromFile += lineFromFile[i];
+        } else {
+            wordNumber++;
+            valueOneLineAssignment(wordNumber, singleWordFromFile);
+            singleWordFromFile = {};
+            std::cout << "\n";
+        }
+    }
+    generateSingleCargo();
+}
+
+void Store::valueOneLineAssignment(const size_t & wordNumber, const std::string & singleWordFromFile) {
+    switch (wordNumber) {
+        case 1 : {
+            oneLine_.className_ = singleWordFromFile;
+        } break;
+        case 2 : {
+            oneLine_.itemName_ = singleWordFromFile;
+        } break;
+        case 3 : {
+            oneLine_.minAmount_ = std::stol(singleWordFromFile); 
+        } break;
+        case 4 : {
+            oneLine_.maxAmount_ = std::stol(singleWordFromFile);
+        } break;
+        case 5 : {
+            oneLine_.minPrice_ = std::stol(singleWordFromFile);
+        } break;
+        case 6 : {
+            oneLine_.maxPrice_ = std::stol(singleWordFromFile);
+        } break;
+        case 7 : {
+            oneLine_.minUniqueStat_ = std::stol(singleWordFromFile);
+        } break;
+        case 8 : {
+            oneLine_.maxUniqueStat_= std::stol(singleWordFromFile);
+        } break;
+        default : {
+        } break;
+    }
```

Comment:
Parsing json will be more user-friendly (avoid numbers, and get real names) whats more you can use one of the multiple external libraries for parsing json, this will make it even easier to write and extend.